### PR TITLE
CORE-9975: Allow `sign` operation to take a key category in context argument

### PR DESCRIPTION
**Pull request by malachyb**

### CODE REVIEW

File: libs/messaging/messaging-impl/src/test/kotlin/net/corda/messaging/publisher/CordaPublisherImplTest.kt [hunk 1]
```diff
@@ -256,7 +256,7 @@ class CordaPublisherImplTest {
     fun testBatchPublishWithMultipleThreads() {
         val publisher = createPublisher(true)
         val numThreads = 100
-        val futures = (0..numThreads).map { CompletableFuture<Unit>() }
+        val futures = List(numThreads) { CompletableFuture<Unit>() }
```
Comment (LWogan) [inline libs/messaging/messaging-impl/src/test/kotlin/net/corda/messaging/publisher/CordaPublisherImplTest.kt]: This change already exists in release 5.1 from this merge. https://github.com/corda/corda-runtime-os/pull/4497/files

I think you need to merge down from release branch

Comment (malachyb) [reply libs/messaging/messaging-impl/src/test/kotlin/net/corda/messaging/publisher/CordaPublisherImplTest.kt]: Merge from release done, not sure how that ended up there. Clean now

File: libs/crypto/cipher-suite/src/main/kotlin/net/corda/crypto/cipher/suite/SigningWrappedSpec.kt [hunk 2]
```diff
@@ -11,6 +11,7 @@ import java.security.PublicKey
  * @param keyMaterialSpec The spec for the wrapped key.
  * @param keyScheme The scheme for the key used for signing operation.
  * @param signatureSpec The signature spec to use for signing, such as SHA256withECDSA, etc.
+ * @param category The category of the key, if specified
```
Comment (dickon) [inline libs/crypto/cipher-suite/src/main/kotlin/net/corda/crypto/cipher/suite/SigningWrappedSpec.kt]: add .

Comment (dickon) [reply libs/crypto/cipher-suite/src/main/kotlin/net/corda/crypto/cipher/suite/SigningWrappedSpec.kt]: thanks, done.

**Review by dickon — APPROVED** [review]
LGTM

**Comment by dickon** [comment]
In particular, now that the API is frozen I think a `cateogry` key in the `context` `Map<String,String>` parameter  is the best way to communicate a category.

**Pull request merged**